aptos-move/framework/aptos-framework/sources/primary_store.move

```diff
@@ -0,0 +1,156 @@
+/// This defines the module for interacting with primary stores of accounts/objects, which have deterministic addresses
+module aptos_framework::primary_store {
+    use aptos_framework::fungible_asset::{Self, FungibleAsset, FungibleAssetStore};
+    use aptos_framework::object::{Self, Object, ConstructorRef, DeriveRef};
+
+    use std::signer;
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Resource stored on the fungible asset metadata object to allow creating primary stores for it.
+    struct DeriveRefPod has key {
```

Comment:
Why is DeriveRefPod a better name?

Reply:
Because it can be used for other purpose. If you agree that function is friend function, I agree to change it back to the original name. Otherwise, it is just a holder of derived ref.

Reply:
@gregnazario Any naming suggestion here? :P

Reply:
I mean I would make it PrimaryWalletCapability, though I think we overloaded that word now...

PrimaryWalletDeriveRef

Tells you what it is, and what it's used for.  Do you suspect it'll be used for anything else in the future?  Then

FungibleAssetDeriveRef